src/main/java/eu/sshoc/dataversesuperset/DataSaver.java

```diff
@@ -72,25 +67,15 @@ public String createTable(DataInfo dataInfo) throws IOException {
 		String insertValues = "INSERT INTO " + tableName + " VALUES ("
 				+ dataInfo.columns.stream().map(c -> "?").collect(Collectors.joining(", "))
 				+ ")";
-		try (CloseableHttpResponse response = dataLoader.openCsv(dataInfo);
-				CSVParser csv = dataLoader.createParser(response.getEntity())) {
-			Iterator<CSVRecord> it = csv.iterator();
-			List<Object[]> rows = new ArrayList<>();
-			int batchLimit = 9000;
-			int columnCount = dataInfo.columns.size();
-			while (it.hasNext()) {
-				CSVRecord record = it.next();
-				Object[] row = new Object[columnCount];
-				for (int i = 0; i < columnCount; i++) {
-					ColumnType type = dataInfo.columns.get(i).type;
-					row[i] = DataInfo.VALUE_PARSERS.get(type).parse(record.get(i));
-				}
-				rows.add(row);
-				if (rows.size() % batchLimit == 0 || !it.hasNext()) {
-					jdbcTemplate.batchUpdate(insertValues, rows);
-				}
-			}
+
+		List<Object[]> rows = dataInfo.reader.createDBInserts(dataInfo, dataLoader);
+
+		int batchLimit = 9000;
+		for(int i = batchLimit; i < rows.size(); i+=batchLimit) {
+			jdbcTemplate.batchUpdate(insertValues, rows.subList(0, i));
 		}
+		jdbcTemplate.batchUpdate(insertValues, rows);
```

Comment:
There was a bug here previously (missing `rows.clear()` after batch update) and it's still here. With over 9000 rows the same data will be inserted multiple times.
Need to make sure every 9000-long batch is inserted only once.

The general idea here was to send rows to the database in parts, without needing to read all the data from input stream into memory. So the `Reader` will need to provide an iterator, instead of full rows list. But that can wait for later if it's too complicated